docs/bugs.json

```diff
@@ -1,4 +1,17 @@
 [
+    {
+        "uid": "SOL-2022-7",
+        "name": "StorageWriteRemovalBeforeConditionalTermination",
+        "summary": "Calling Yul functions that conditionally terminate the transaction using ``return(...)`` or ``stop()`` may result in incorrect removals of storage writes.",
+        "description": "A call to a Yul function that conditionally terminates the transaction did not correctly mark all prior changes to storage as 'used', which may result in storage writes to be incorrectly removed. This happens in cases in which it had been valid to remove the store, if the Yul function in question never actually terminated the transaction. Conditional termination of the transaction within the same Yul function is not affected. In Solidity with via-IR code generation, any state change before a function conditionally calling ``return(...)`` or ``stop()`` in inline assembly, may be incorrectly removed, whenever it had been valid to remove the change without the ``return(...)`` or ``stop()``. Since the bug only occurs on calls to Yul functions, inlining may prevent the bug in some cases.",
```

Comment:
```suggestion
        "description": "A call to a Yul function that conditionally terminates the transaction did not correctly mark all prior changes to storage as *used*, which may result in storage writes being incorrectly removed. This happens in cases in which it would have been valid to remove the store, if the Yul function in question never actually terminated the transaction. Conditional termination of the transaction within the same Yul function is not affected. In Solidity with via-IR code generation, any state change before a function conditionally calling ``return(...)`` or ``stop()`` in inline assembly, may be incorrectly removed, whenever it would have been valid to remove the change without the ``return(...)`` or ``stop()``. Since the bug only occurs on calls to Yul functions, inlining may prevent the bug in some cases.",
```